packages/x-data-grid/src/hooks/features/listView/useGridListView.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import type { GridStateColDef } from '../../../models/colDef/gridColDef';
+import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridPrivateApiCommunity } from '../../../models/api/gridApiCommunity';
+import { GridListViewApi } from '../../../models/api/gridListViewApi';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridEventListener } from '../../../models/events';
+import {
+  gridListColumnSelector,
+  gridVisibleListColumnDefinitionsSelector,
+} from './gridListViewSelectors';
+import { gridColumnDefinitionsSelector, gridVisibleColumnDefinitionsSelector } from '../columns';
+import { gridDimensionsSelector } from '../dimensions';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+
+export type GridListViewState = { listColumn: GridStateColDef | undefined };
+
+const getListColumnWidth = (apiRef: React.MutableRefObject<GridPrivateApiCommunity>) => {
+  const dimensions = gridDimensionsSelector(apiRef.current.state);
+  const columns = gridVisibleColumnDefinitionsSelector(apiRef);
+  const actionsColumn = columns.find((col) => col.type === 'actions');
+  const viewportWidth = dimensions.viewportInnerSize.width;
+  const listColumnWidth = actionsColumn
+    ? viewportWidth - actionsColumn.computedWidth
+    : viewportWidth;
+  return listColumnWidth;
+};
+
+export const listViewStateInitializer: GridStateInitializer<
+  Pick<DataGridProcessedProps, 'unstable_listColumn'>
+> = (state, props, apiRef) => ({
+  ...state,
+  listView: {
+    listColumn: { ...props.unstable_listColumn, computedWidth: getListColumnWidth(apiRef) },
+  },
+});
+
+export function useGridListView(
+  apiRef: React.MutableRefObject<GridPrivateApiCommunity>,
+  props: Pick<DataGridProcessedProps, 'unstable_listColumn'>,
+) {
+  /**
+   * API METHODS
+   */
+  const getListColumn = React.useCallback<GridListViewApi['getListColumn']>(
```

Comment:
I would argue against using `React.useCallback` unless necessary, it makes for very verbose lines and has a runtime cost.

Reply:
Fixed https://github.com/mui/mui-x/pull/14393/commits/9b5dc777397944624b518b2fcf6b8a8911fa7fe3